lib/datadog/tracing/contrib/rails/configuration/settings.rb

```diff
@@ -24,30 +24,35 @@ def initialize(options = {})
             end
 
             option :enabled do |o|
-              o.default { env_to_bool(Ext::ENV_ENABLED, true) }
+              o.type :bool
+              o.env Ext::ENV_ENABLED
+              o.default true
             end
 
             option :analytics_enabled do |o|
-              o.default { env_to_bool(Ext::ENV_ANALYTICS_ENABLED, nil) }
-
+              o.type :bool, nilable: true
+              o.env Ext::ENV_ANALYTICS_ENABLED
               o.on_set do |value|
                 # Update ActionPack analytics too
-                Datadog.configuration.tracing[:action_pack][:analytics_enabled] = value
+                Datadog.configuration.tracing[:action_pack][:analytics_enabled] = value unless value.nil?
               end
```

Comment:
It's subtle, but since both this setting, as well as the action_pack setting allow `nil`, I think perhaps that `unless value.nil?` is perhaps not the best idea (because in a way, `nil` previously meant "disable everything" and now only means "disable only rails".

In practice in most cases it'll be equivalent, but my thinking is, why change it, if we can keep the exact same previous behavior?

Reply:
Done [00b1168](https://github.com/DataDog/dd-trace-rb/pull/2988/commits/00b1168187a12a8bc75e8a65a7063f3845e6c3b1)

On a private Zoom call, we also discussed that `rails` and `action_pack` supporting `nil` for `analytics_enabled` is weird and that we should use a boolean value instead.

That change should be done on a separate PR